File: Flora frontend/flora_mobile_app/lib/models/donation.dart
class Donation {
  final int? id;
  final String donorName;
  final String email;
  final double amount;
  final String purpose;
  final DateTime? date;
  final int campaignId;
  final int? userId;
  final String? campaignTitle;
  final String? status;

  Donation({
    this.id,
    required this.donorName,
    required this.email,
    required this.amount,
    required this.purpose,
    this.date,
    required this.campaignId,
    this.userId,
    this.campaignTitle,
    this.status,
  });

  factory Donation.fromJson(Map<String, dynamic> json) {
    return Donation(
      id: json['id'],
      donorName: json['donorName'],
      email: json['email'],
      amount: json['amount']?.toDouble() ?? 0.0,
      purpose: json['purpose'],
      date: json['date'] != null ? DateTime.parse(json['date']) : null,
      campaignId: json['campaignId'],
      userId: json['userId'],
      campaignTitle: json['campaignTitle'],
      status: json['status'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'donorName': donorName,
      'email': email,
      'amount': amount,
      'purpose': purpose,
      'campaignId': campaignId,
      'userId': userId,
      'status': status,
    };
  }
}

File: Flora backend/Flora.Services/Database/Donation.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Flora.Services.Database
{
    public class Donation
    {
        public int Id { get; set; }
        public int UserId { get; set; }
        public User? User { get; set; }
        public string DonorName { get; set; } = string.Empty;
        public string Email { get; set; } = string.Empty;
        public double Amount { get; set; }
        public string Purpose { get; set; } = string.Empty;
        public DateTime Date { get; set; } = DateTime.Now;
        public string? TransactionId { get; set; }
        public string Status { get; set; } = "Pending";

        public int CampaignId { get; set; }
        public virtual DonationCampaign? Campaign { get; set; }
    }
}
